src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/LoggingPolicy.cs

```diff
@@ -38,20 +39,27 @@ public override async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMem
             var after = Stopwatch.GetTimestamp();
 
             var status = message.Response.Status;
-            // if error status
-            if (status >= 400 && status <= 599 && (Array.IndexOf(_excludeErrors, status) == -1))
+            bool isError = status >= 400 && status <= 599 && (Array.IndexOf(_excludeErrors, status) == -1);
+            bool wrapResponseStream = s_eventSource.ShouldLogContent(isError) && message.Response.ResponseContentStream?.CanSeek == false;
+
+            if (wrapResponseStream)
+            {
+                message.Response.ResponseContentStream = new LoggingStream(message.Response.RequestId, s_eventSource, message.Response.ResponseContentStream, isError);
+            }
+
+            if (isError)
             {
                 s_eventSource.ErrorResponse(message.Response);
 
-                if (message.Response.ResponseContentStream != null)
+                if (!wrapResponseStream && message.Response.ResponseContentStream != null)
                 {
                     await s_eventSource.ErrorResponseContentAsync(message.Response, message.Cancellation).ConfigureAwait(false);
```

Comment:
nit:  The name `ErrorResponseContentAsync` feels awkward to me.  It may be more clear if it was something like `ReceivedErrorResponseContentAsync`  or `ReportErrorResponseContentAsync` or similar.

Reply:
I'm trying to keep method names in sync with event names (barring the Async suffix)